pkg/skaffold/build/gcb/cloud_build.go

```diff
@@ -381,3 +369,44 @@ func (b *Builder) createBucketIfNotExists(ctx context.Context, c *cstorage.Clien
 	log.Entry(ctx).Debugf("Created bucket %s in %s", bucket, projectID)
 	return nil
 }
+
+func (b *Builder) createCloudBuild(ctx context.Context, cbclient *cloudbuild.Service, projectID string, buildSpec cloudbuild.Build) (
+	string, func(opts ...googleapi.CallOption) (*cloudbuild.Build, error), error) {
+	var op *cloudbuild.Operation
+	var err error
+	if b.WorkerPool == "" {
+		op, err = cbclient.Projects.Builds.Create(projectID, &buildSpec).Context(ctx).Do()
+		if err != nil {
+			return "", nil, sErrors.NewErrorWithStatusCode(&proto.ActionableErr{
+				ErrCode: proto.StatusCode_BUILD_GCB_CREATE_BUILD_ERR,
+				Message: fmt.Sprintf("error creating build: %s", err),
+			})
+		}
+		remoteID, errB := getBuildID(op)
+		if errB != nil {
+			return "", nil, sErrors.NewErrorWithStatusCode(&proto.ActionableErr{
+				ErrCode: proto.StatusCode_BUILD_GCB_GET_BUILD_ID_ERR,
+				Message: err.Error(),
+			})
+		}
+		return remoteID, cbclient.Projects.Builds.Get(projectID, remoteID).Do, nil
+	}
+	location := strings.Split(b.WorkerPool, "/workerPools/")[0]
```

Comment:
This feels like something that should be in a helper function.

There's a section in this file where we attempt to guess the project from the image ref; we should also try using this workerPool string too.

Reply:
I am not sure, how it works but you could create a build in a public workerpool created in another project. Hence did not touch that code.